meta-client/src/main/java/org/apache/helix/metaclient/recipes/lock/LockClient.java

```diff
@@ -0,0 +1,112 @@
+package org.apache.helix.metaclient.recipes.lock;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+import org.apache.helix.metaclient.api.MetaClientInterface;
+import org.apache.helix.metaclient.api.Op;
+import org.apache.helix.metaclient.datamodel.DataRecord;
+import org.apache.helix.metaclient.exception.MetaClientException;
+import org.apache.helix.metaclient.factories.MetaClientConfig;
+import org.apache.helix.metaclient.factories.MetaClientFactory;
+import org.apache.helix.metaclient.impl.zk.factory.ZkMetaClientConfig;
+import org.apache.helix.metaclient.impl.zk.factory.ZkMetaClientFactory;
+import org.apache.helix.zookeeper.datamodel.serializer.ZNRecordSerializer;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class LockClient implements LockClientInterface, AutoCloseable {
+  private final MetaClientInterface<LockInfo> _metaClient;
+
+  public LockClient(MetaClientConfig config) {
+    if (config.getStoreType() == MetaClientConfig.StoreType.ZOOKEEPER) {
```

Comment:
Enum suggest to use equals. Also better impl like:

MetaClientConfig.StoreType.ZOOKEEPER.equals(config.getStoreType()) 

to avoid NPE


Also for entire piece of logic, it would be better:

if (!MetaClientConfig.StoreType.ZOOKEEPER.equals(config.getStoreType())) {
   throw new MetaClientException("Unsupported store type: " + config.getStoreType());
}

xxxx // do your logic.


This does not require many if else clauses.

Reply:
@junkaixue I have the opposite opinion on Enum compare.
Using `==` is completely fine, actually `Enum::equals()` directly calls to `==`, it avoid NPE too.

Reply:
Yeah. Good deep analysis. In usual pattern Enum equals is calling the "==", but for some overrides enums, especially it has internal property ones, equals can be override. When we do coding, it would be good not making assumption of it is native Enum class.

The comment is more leaning to let Marcos learn some Java coding conventions.

Reply:
Thanks @junkaixue and @qqu0127 for the input! Even though it is essentially the same, I'll change it to enum compare for good practices.